Document/0x05b-Basic-Security_Testing.md

```diff
@@ -366,7 +366,109 @@ More infomation on using the Objection REPL can be found on the [Objection Wiki]
 
 ##### radare2
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1231>
+[radare2](https://rada.re/r/ "Radare2 official website") is a popular reverse engineering framework for analysis, disassembling, debugging and patching that is scriptable and supports many architectures and file formats including Android/iOS apps. For Android, Dalvik DEX (odex, multidex), ELF (executables, .so, ART) and Java classes are supported, and it even has a web server for collaborative work. It also contains several useful scripts that can help you during mobile application analysis as it offers low level disassembling and safe static analysis that comes handy when traditional tools fail.
+
+###### Installing radare2
+
+Please refer to [radare2's official installation instructions](https://github.com/radare/radare2/blob/master/README.md "radare2 installation instructions"). There are also installation instructions If you need to [install radare2 on Android](https://github.com/radare/radare2/blob/master/doc/android.md "Install radare2 on Android").
+
+###### Using radare2
+
+You can use `rafind2` to read strings directly from the encoded AndroidManifest.xml. Use this to list permissions, activities, providers, services, receivers and other information stored in AndroidManifest.xml
+
+```shell
+$ rafind2 -ZS permission AndroidManifest.xml
+```
+
+```shell
+$ rafind2 -ZS activity AndroidManifest.xml
+```
+
+```shell
+$ rafind2 -ZS provider AndroidManifest.xml
+```
+
+```shell
+$ rafind2 -ZS service AndroidManifest.xml
+```
+
+```shell
+$ rafind2 -ZS receiver AndroidManifest.xml
+```
+
+You can use the flag `-a` to manually set the architecture to Dalvik.
```

Comment:
Explain briefly here/above what's rafind2 and r2 and why you use it this way. Also you'd like to use rabin2 for some quick things and r2 to go deeper? We don't have to give much detail here. It should be explained properly in the Rev. Eng. chapter.

Reply:
Done.